### PR TITLE
user interface settings, font-size

### PR DESCRIPTION
**Pull request by iqmulator**

### CODE REVIEW

File: src/nodejs/src/font-size-init.ts [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+const FontSize : {[title: string]: string} = {
+    'Small': '14px',
+    'Medium': '16px',
+    'Large': '18px',
+    'XL': '20px',
+    'XXL': '24px',
+}
+
+export class FontSizeInit {
+    private static fontSize= FontSize['Medium'];
```
Comment (crui3er) [inline src/nodejs/src/font-size-init.ts]: missing space between 'fontSize' and '='

File: src/nodejs/src/font-size-init.ts [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+            localStorage.setItem('font-size', this.fontSize);
+        }
+        if (rootFontSize != this.fontSize) {
+            (root as HTMLElement).style.setProperty('--font-size', this.fontSize);
+        }
+
+        ScreenSize.change$.subscribe(_ => this.init());
+    }
+
+    private static isLocalFontSizeNull() {
```
Comment (crui3er) [inline src/nodejs/src/font-size-init.ts]: Rename to something like 'setDefaultFontSize'

File: src/nodejs/src/font-size-init.ts [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+            this.fontSize = FontSize['Large'];
+        } else {
+            this.fontSize = FontSize['Medium'];
+        }
+    }
+
+    private static checkLocalFontSizeValues(fontValue: string) : boolean {
+        let result = false;
+        Object.values(FontSize).forEach(v => {
+            console.log('v: ', v);
```
Comment (crui3er) [inline src/nodejs/src/font-size-init.ts]: remove console output here

File: src/nodejs/src/font-size-init.ts [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+
+    private static isLocalFontSizeNull() {
+        if (DeviceInfo.isIos) {
+            this.fontSize = FontSize['Large'];
+        } else {
+            this.fontSize = FontSize['Medium'];
+        }
+    }
+
+    private static checkLocalFontSizeValues(fontValue: string) : boolean {
```
Comment (crui3er) [inline src/nodejs/src/font-size-init.ts]: rename to 'isValidFontSizeValue'

File: src/nodejs/src/font-size-init.ts [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+        let result = false;
+        Object.values(FontSize).forEach(v => {
+            console.log('v: ', v);
+            if (v == fontValue)
+                result = true;
+        });
+        return result;
+    }
+
+    private static reverseFontSize() {
```
Comment (crui3er) [inline src/nodejs/src/font-size-init.ts]: better embed this method into 'getRootFontSize', its name is misleading

File: src/nodejs/src/font-size-init.ts [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+        }
+
+        if (localFontSize != this.fontSize) {
+            localStorage.setItem('font-size', this.fontSize);
+        }
+        if (rootFontSize != this.fontSize) {
+            (root as HTMLElement).style.setProperty('--font-size', this.fontSize);
+        }
+
+        ScreenSize.change$.subscribe(_ => this.init());
```
Comment (crui3er) [inline src/nodejs/src/font-size-init.ts]: Why do you need to run init again on size change?

Comment (crui3er) [inline src/nodejs/src/font-size-init.ts]: There is an issue here, that every screen size change will add one more subscription on screen size change.
You can set a flag that you subscribed already and next time skip subscription.
Or refactor code to something like:
public static init(): void {
    this.initInternal();
    ScreenSize.change$.subscribe(_ => this.initInternal());
}

**Review by crui3er — CHANGES_REQUESTED** [review]

File: src/nodejs/src/font-size-init.ts (hunk 6 above)
Comment (iqmulator) [reply src/nodejs/src/font-size-init.ts]: This is what I asked you to pay attention to, thank you))

Comment (iqmulator) [reply src/nodejs/src/font-size-init.ts]: done

**Pull request merged**